spacy/pipeline/coref.py

```diff
@@ -511,15 +509,20 @@ def update(
         set_dropout_rate(self.model, drop)
 
         total_loss = 0
-
-        for eg in examples:
-            preds, backprop = self.model.begin_update([eg.predicted])
-            score_matrix, mention_idx = preds
-
-            loss, d_scores = self.get_loss([eg], score_matrix, mention_idx)
+        docs = [eg.predicted for eg in examples]
+        for doc, eg in zip(docs, examples):
+            # replicates the EntityLinker's behaviour and
+            # copies annotations over https://bit.ly/3iweDcW
+            # takes 'coref_head_clusters' from the reference.
+            for key, sg in eg.reference.spans.items():
+                if key.startswith(self.input_prefix):
+                    aligned_spans = eg.get_aligned_spans_x2y(sg)
+                    doc.spans[key] = [doc[span.start:span.end] for span in aligned_spans]
```

Comment:
I think this can just be:

```
doc.spans[key] = eg.get_aligned_spans_y2x(sg)
```

Note this is `y2x` and not `x2y`.

Reply:
Another thing is that this because this unconditionally overwrites data on the `predicted` Doc, it modifies that Doc for other components in the pipeline. To avoid that it's necessary to save the state of the Doc (`doc.spans` in this case) and restore it afterwards. Check how `old_ents` is used in the EntityLinker. In this case you can probably just do something like:

```
old_spans = [eg.predicted.spans for eg in examples]
# and to reset...
for spans, eg in zip(old_spans, examples):
    eg.predicted.spans = spans
```